pkg/datapath/loader/compile.go

```diff
@@ -284,6 +290,9 @@ func compile(ctx context.Context, prog *progInfo, dir *directoryInfo) (err error
 
 	args = append(args, standardCFlags...)
 	args = append(args, progCFlags(prog, dir)...)
+	if len(prog.ExtraCArgs) != 0 {
+		args = append(args, prog.ExtraCArgs...)
+	}
```

Comment:
You don't need to do this: https://play.golang.org/p/qRRZnHEYStM

```suggestion
	args = append(args, prog.ExtraCArgs...)
```